lib/internal/modules/cjs/loader.js

```diff
@@ -303,19 +303,19 @@ function readPackage(requestPath) {
 
 function readPackageScope(checkPath) {
   const rootSeparatorIndex = checkPath.indexOf(path.sep);
-  let separatorIndex;
-  while (
-    (separatorIndex = checkPath.lastIndexOf(path.sep)) > rootSeparatorIndex
-  ) {
-    checkPath = checkPath.slice(0, separatorIndex);
+  let separatorIndex = checkPath.lastIndexOf(path.sep);
+  do {
     if (checkPath.endsWith(path.sep + 'node_modules'))
       return false;
     const pjson = readPackage(checkPath);
```

Comment:
Hmm now that I think about it it looks like this is applying `checkPath` with the input filename, `file.js`, resulting in a stat to `/path/to/file.js/package.json` which doesn't seem good.

Reply:
`checkPath` is just a String object here. Unsure how it ever gets applied.

Reply:
`checkPath` when this function is called is a direct path to an exact file (the parent path), like `/path/to/file.js`. This logic would try to check the package.json (via `readPackage`) of `/path/to/file.js/package.json` which is a call that shouldn't be necessary.

Reply:
I noticed some strange behavior as you describe, which is what lead me to do the special-casing described in https://github.com/nodejs/node/pull/34595#discussion_r469626073.